test-data/unit/fine-grained-suggest.test

```diff
@@ -1090,7 +1090,7 @@ optional2(10)
 optional2('test')
 
 def optional3(x: Optional[List[Any]]):
-    assert not x
+    assert x
```

Comment:
```suggestion
    assert not x and x is not None
```
This would work as well, although as mentioned earlier, I'm not convinced it's better. Feel free to accept or dismiss it.